pootle/apps/pootle_app/management/commands/create_project.py

```diff
@@ -0,0 +1,153 @@
+            default="localfs",
+            choices=fs_plugins.gather().keys(),
+            help="Filesystem type: {0}".format(
+                " | ".join(fs_plugins_names)),
+        )
+        parser.add_argument(
+            "--fs-url",
+            action="store",
+            dest="fs_url",
+            default="",
+            help="Filesystem path or URL.",
+        )
+        parser.add_argument(
+            "--source-language",
+            action="store",
+            dest="sourcelang",
+            default="en",
+            help="Source language. Examples: [en | es | fr | ...].\
+                     Default: %(default)s",
+        )
+        parser.add_argument(
+            "--report-email",
+            action="store",
+            dest="contact",
+            help="Contact email for reports. Example: admin@mail.com. Default: ",
+        )
+        parser.add_argument(
+            "--disabled",
+            action="store_true",
+            dest="disabled",
+            help="Does the project start disabled?",
+        )
+        parser.add_argument(
+            "--checkstyle",
+            action="store",
+            dest="checkstyle",
+            choices=PROJECT_CHECKERS.keys(),
+            default="standard",
+            help="Quality check styles. Example: {0}. Default: %(default)s"
+                 .format(" | ".join(PROJECT_CHECKERS.keys())),
+        )
+
+    @transaction.atomic
+    def handle(self, **options):
+        """Immitates the behaviour of the admin interface
+            for creating a project from command line."""
+
+        try:
+            source_language_obj = Language.objects.get(code=options["sourcelang"])
+        except Language.DoesNotExist:
+            raise CommandError("Source language %s does not exist"
+                               % options["sourcelang"])
+
+        if options["contact"]:
+            try:
+                validate_email(options["contact"])
+            except ValidationError as e:
+                raise CommandError(e)
+
+        new_project = Project.objects.create(
+            code=options["code"],
+            fullname=options["name"] or options["code"].capitalize(),
+            checkstyle=options["checkstyle"],
+            source_language=source_language_obj,
+            filetypes=options["filetypes"] or ["po"],
+            report_email=options["contact"] or "",
+            disabled=options["disabled"])
+
+        if options["preset_mapping"]:
+            new_project.config["pootle_fs.translation_mapping"] = dict(
+                default=FS_PRESETS[options["preset_mapping"]][0])
+        else:
+            new_project.config["pootle_fs.translation_mapping"] = dict(
+                default=options["mapping"])
+
+        new_project.config["pootle_fs.fs_type"] = options["fs_type"]
+
+        if options["fs_url"]:
+            try:
+                new_project.config["pootle_fs.fs_url"] = options["fs_url"]
+            except ValidationError as e:
```

Comment:
I still think that all input should be validated before doing any change. That way we can avoid the situation where a project gets created but its config is not properly set. Have in mind this command only allows to create new projects, but not altering already existing projects.

Also if we can verify beforehand if the input is wrong or not, why not just tell the user to fix the mistakes before committing any change?